tableauserverclient/models/connection_credentials.py

```diff
@@ -30,11 +35,11 @@ def oauth(self):
 
     @oauth.setter
     @property_is_boolean
-    def oauth(self, value):
+    def oauth(self, value: bool):
         self._oauth = value
 
     @classmethod
-    def from_xml_element(cls, parsed_response, ns):
+    def from_xml_element(cls, parsed_response: ET.ElementTree, ns: Dict) -> T:
```

Comment:
There's something going on here, when I run mypy on your PR, there are a bunch of complaints about ElementTree not having a `get` method. I think that's because those are `Element` not `ElementTree` but I thought it should be smart enough to infer that is what `.find` returns?

Reply:
I'm not entirely sure what is going on here. I admittedly should have started with mypy, but instead I did it by looking through some of my application code. Mypy complains about a wide array of things when I try to run it, and it will take some time to sort out. I wanted to get a PR up, perhaps too early, to get a reading on whether or not this kind of thing would be accepted. Seems like so if I can iron out the issues in it.

Reply:
Now that we only support 3.5 and above, I would be open to type hints for sure.

Maybe we want to start with just 1 vertical, like DatasourceItem, DatasourceEndpoints, and use that vertical to sort out some of the general patterns we may need (like Element/ElementTree) or if we want type_checking IF statements etc

Reply:
Yeah, I agree with Tyler on starting out with 1 vertical slice.
Type hints would be great to have!